app/routes.py

```diff
@@ -1,2 +1,245 @@
+            }
+    })
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def handle_task_incomplete(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404)
+
+    task.completed_at = None
+
+    db.session.commit()
+
+    return make_response({
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False
+            }
+    })
+
+@goals_bp.route("", methods=["POST", "GET"])
+def handle_goals():
+        if request.method == "GET":
+            goals = Goal.query.all()
+            goals_response = []
+            for goal in goals:
+                goals_response.append({
+                    "id": goal.goal_id,
+                    "title": goal.title
+                })
+            sort_by_title = request.args.get("sort")
+            if sort_by_title:
+                if sort_by_title == "asc":
+                    goals_response = sorted(goals_response, key = lambda i: i['title'])
+                if sort_by_title == "desc":
+                    goals_response = sorted(goals_response, key = lambda i: i['title'],reverse=True)
+            return jsonify(goals_response), 200
+
+        elif request.method == "POST":
+            request_body = request.get_json()
+            if 'title' not in request_body:
+                return {"details": "Invalid data"}, 400
+            new_goal = Goal(title=request_body["title"])
+
+            db.session.add(new_goal)
+            db.session.commit()
+
+        return {
+            "goal": {
+                "id": new_goal.goal_id,
+                "title": new_goal.title
+            }
+        }, 201
+
+@goals_bp.route("/<goal_id>", methods=["GET", "DELETE", "PUT"])
+def handle_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        return make_response("", 404)
+
+    if request.method == "GET":
+        return {
+            "goal":{
+                "id": goal.goal_id,
+                "title": goal.title
+            }
+        }
+    elif request.method == "DELETE":
+        message = {"details": f"Goal {goal.goal_id} \"{goal.title}\" successfully deleted"}
+        db.session.delete(goal)
+        db.session.commit()
+        return make_response(message)
+    elif request.method == "PUT":
+        form_data = request.get_json()
+
+        goal.title = form_data["title"]
+
+        db.session.commit()
+
+        return make_response({
+                "goal": {
+                    "id": goal.goal_id,
+                    "title": goal.title
+                }
+        })
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST", "GET"])
+def handle_goal_tasks(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        return make_response("", 404)
+
+    if request.method == "GET":
+        tasks = Task.query.filter(Task.goal_id == goal_id)
```

Comment:
We can retrieve the tasks this way, but also, our Goal model has a tasks member, so we could also do
```python
tasks = goal.tasks
```